### PR TITLE
Add more core imports to extend

### PR DESCRIPTION
**Pull request by chaserileyroberts**
Adds many of the types needed when interpreting a jaxpr.

### CODE REVIEW

**Review by froystig — APPROVED** [review]

File: jax/extend/core.py [hunk 1]
```diff
@@ -16,5 +16,16 @@
 # See PEP 484 & https://github.com/google/jax/issues/7570
 
 from jax._src.abstract_arrays import (
-  array_types as array_types
+  array_types as array_types,
+)
+from jax._src.core import (
+    ClosedJaxpr as ClosedJaxpr,
+    eval_jaxpr as eval_jaxpr,
```
[the remark this answers is not in the document]
Comment (chaserileyroberts) [reply jax/extend/core.py:23]: Yeah that is likely better for everyone in the long term lol.

[the remark this answers is not in the document]
Comment (chaserileyroberts) [reply jax/extend/core.py:23]: Done

**Review by froystig — APPROVED** [review]

**Comment by chaserileyroberts** [comment]
What's the status of this?

**Comment by froystig** [comment]
Build issues that we need to fix up before merging.

**Pull request closed without merging**